core/src/test/java/org/apache/iceberg/TestUpdatePartitionSpec.java

```diff
@@ -614,6 +605,65 @@ public void testRemoveAndAddMultiTimes() {
     V2Assert.assertEquals("Should match expected spec", v2Expected, updated);
   }
 
+  @Test
+  public void testRemoveAndUpdateWithIdentity() {
```

Comment:
I'm not sure that this is a valid case. It is currently allowed by the updated implementation, but this is a case that we originally wanted to prevent with the check for adding a deleted field. I think that the update should only allow adding a field with the same name if it uses a different transform.

Reply:
Hi, @rdblue I know this is kind of a behavior change in iceberg, but I am wondering what is the benefit for the previous behavior ?

I just found it convenient for us to lift the original restriction as we can blindly remove conflicted column names and add new instead of adding more checks if no significant benefit there

Reply:
I don't think this is a good idea. You don't want to create a new partition spec when there are no real changes. If you want to do this, then I suggest opening a separate PR that detects that one of the existing changes was undone (removing field "ts") and removes the change rather than performing a removal and an add.

Reply:
@rdblue sure, I just removed the test and recover the original one